src/tds/codec/token/token_row.rs

```diff
@@ -1,21 +1,88 @@
-use crate::{tds::codec::ColumnData, SqlReadBytes};
+mod into_row;
+use crate::{tds::codec::ColumnData, MetaDataColumn, SqlReadBytes, TokenType};
+use asynchronous_codec::BytesMut;
+use bytes::BufMut;
 use futures::io::AsyncReadExt;
+pub use into_row::IntoRow;
 
-#[derive(Debug)]
-pub struct TokenRow {
-    data: Vec<ColumnData<'static>>,
+/// A row of data.
+#[derive(Debug, Default, Clone)]
+pub struct TokenRow<'a> {
+    data: Vec<ColumnData<'a>>,
 }
 
-impl IntoIterator for TokenRow {
-    type Item = ColumnData<'static>;
+impl<'a> IntoIterator for TokenRow<'a> {
+    type Item = ColumnData<'a>;
     type IntoIter = std::vec::IntoIter<Self::Item>;
 
     fn into_iter(self) -> Self::IntoIter {
         self.data.into_iter()
     }
 }
 
-impl TokenRow {
+impl<'a> TokenRow<'a> {
+    pub(crate) fn encode(
```

Comment:
We can still use the trait. Please do so, as I described in the previous comments.

Reply:
taken